### PR TITLE
Use Make Automatic Variables, update msvs-detect

### PR DESCRIPTION
**Pull request by MisterDA**
Hacking in passing, I decided to use Make Automatic Variables to avoid redundancies with file names, and update msvc-detect.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -91,6 +93,11 @@ MSVCC64 = cl.exe $(MSVC_FLAGS)
 endif
 endif
 
+ifeq ($(shell expr "$(CI)" : '^[Tt]rue$$'),4)
```
Comment (dra27) [inline Makefile:96]: I think this avoids the shell call (which matters a lot to me... cf. #119)
```suggestion
ifneq "$(filter True true,$(CI))" ""
```

**Review by dra27 — COMMENTED** [review]
In-passing comment on one of the Makefile changes!

I've looked at some of the earlier commits of this already - I like the direction it's going, but could you split the first 3 commits to a separate PR because I think they should all pass immediately and it will get the noise out of the way of looking at the C compiler warnings part?

**Comment by MisterDA** [comment]
I've removed the compiler warnings commits from this PR.

**Review by dra27 — APPROVED** [review]
Lovely, ta!

**Pull request merged**